src/spec.ts

```diff
@@ -101,8 +101,28 @@ export type Spec = UnitSpec | FacetSpec | LayerSpec;
 
 /* Custom type guards */
 
-export function isFacetSpec(spec: ExtendedSpec): spec is FacetSpec {
-  return spec['facet'] !== undefined;
+export function isFacetSpec(spec: ExtendedSpec | ExtendedFacetSpec): spec is FacetSpec {
```

Comment:
Ok, so I will write something like `isSomeFacetSpec(spec: ExtendedSpec): spec is FacetSpec | ExtendedFacetSpec` ?